src/utils/dif_upload.rs

```diff
@@ -273,19 +273,22 @@ impl AsRef<[u8]> for DifMatch<'_> {
     }
 }
 
-trait ToAssemble {
-    fn to_assemble(&self, with_debug_id: bool) -> (Digest, ChunkedDifRequest<'_>);
+trait IntoAssembleRequestTuple<'a> {
```

Comment:
The name is a mouthful. You could call it `IntoAssembleRequest` and define `AssembleRequest` as a type alias for the tuple, or a proper struct.

Also, I believe this becomes much nicer if you implement the method as taking `&self` and find a different name for it that `clippy` will accept.